source/common/upstream/cluster_manager_impl.h

```diff
@@ -209,11 +210,11 @@ class ClusterManagerImpl : public ClusterManager, Logger::Loggable<Logger::Id::u
 
   struct PrimaryClusterData {
     PrimaryClusterData(uint64_t config_hash, bool added_via_api, ClusterPtr&& cluster)
-        : config_hash_(config_hash), added_via_api_(added_via_api), cluster_(std::move(cluster)) {}
+        : config_hash_(config_hash), added_via_api_(added_via_api), cluster_(cluster.release()) {}
```

Comment:
Done. Figured out the mock issue with shared pointers in expectations I stumbled on earlier.

Alternatively could also use a bare pointer in the factory APIs, which results in somewhat cleaner (test) code.